### DISCUSSION TITLE
graphql middleware

### DISCUSSION THREAD

**Discussion by devmnj** [nuxt/nuxt#17801] [Questions]
In Nuxt 2 it was possible to create graphql servermiddleware api route , how can I do it in Nuxt3?

**Comment by HendrikJan** [comment]
I use `express-graphql` in the `/server/api` folder.
As `express-graphql` uses the same `(req, res) => 'result'` function as Nuxt (unjs/h3) uses, this just works.

I just started working on this, so I don't know yet if everything (like websockets) will work.

File: /server/api/graphql.ts:
```ts
import { graphqlHTTP } from 'express-graphql';
import { makeExecutableSchema } from '@graphql-tools/schema';
import _PrismaClient from '@prisma/client'
const { PrismaClient } = _PrismaClient

const prisma = new PrismaClient()

const typeDefs = `
  type Course {
    code: String!
    name: String!
  }

  type Query {
    allCourses: [Course!]!
  }
`;

const resolvers = {
  Query: {
    allCourses: () => {
      return prisma.course.findMany();
    }
  }
};

export const schema = makeExecutableSchema({
  resolvers,
  typeDefs,
});

export default graphqlHTTP({
  schema,
  graphiql: true,
})
```

You can then access the api at `https://yoursite:3000/api/graphql`.

**Reply by lewebsimple** [reply]
I'm currently trying to get graphql-ws to work with Nuxt3.
The challenge is this is very platform-specific, i.e. what works on a Node server won't work on Cloudflare Workers or Vercel.
